pkg_building.Rmd

```diff
@@ -52,12 +52,29 @@ We recommend you to use the [`codemetar` package](https://github.com/ropensci/co
 
 * Use `message()` and `warning()` to communicate with the user in your functions. Please do not use `print()` or `cat()` unless it's for a `print.*()` or `str.*()` methods, as these methods of printing messages are harder for the user to suppress.
 
+### Input checking
+
+We recommend your package use a consistent method of your choice for [checking inputs](https://blog.r-hub.io/2022/03/10/input-checking/) -- either base R, an R package, or custom helpers.
+
 ### Interactive/Graphical Interfaces
 
 If providing graphical user interface (GUI) (such as a Shiny app), to facilitate workflow, include a mechanism to automatically reproduce steps taken in the GUI.  This could include auto-generation of code to reproduce the same outcomes, output of intermediate values produced in the interactive tool, or simply clear and well-documented mapping between GUI actions and scripted functions. (See also ["Testing"](#testing) below.)
 
 The [`tabulizer` package](https://github.com/ropensci/tabulizer) e.g. has an interactive workflow to extract tables, but can also only extract coordinates so one can re-run things as a script. Besides, two examples of shiny apps that do code generation are https://gdancik.shinyapps.io/shinyGEO/, and https://github.com/wallaceEcoMod/wallace/.
 
+### Packages wrapping web resources (API clients)
+
+If your package accesses a web API or another web resource,
+
+- Make sure requests send an user agent. The users should be able to override the package's default user agent. Ideally the user agent should be different on continuous integration services, and in development (based on, for instance, the GitHub usernames of the developers).
+- You might choose different (better) defaults than the API, in which case you should document them.
+- Your package could help with pagination, by allowing the users to not worry about it at all since your package does all necessary requests.
```

Comment:
```suggestion
- Your package should help with pagination, by allowing the users to not worry about it at all since your package does all necessary requests.
```